### PR TITLE
Added Counting Sort [C#]

### LINKED ISSUE #611: Counting Sort [C#]
#### This issue is about [Counting Sort](http://www.geeksforgeeks.org/counting-sort/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by DimaMukhin**
Fixes #611 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once the Travis build passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: counting_sort/CountingSort.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class CountingSort
+{
+    /// <summary>
+    /// Sorts the array using counting sort
+    /// Time Complexity: O(n + k) where n is the number of elements in the array and k is the range of input
+    /// Space Complexity: O(n + k)
+    /// </summary>
+    /// <param name="arr">The original unsorted array</param>
+    /// <returns>The sorted array</returns>
```
Comment (aashutoshrathi) [inline counting_sort/CountingSort.cs]: Make it one block comment instead.

File: counting_sort/CountingSort.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        for (int i = 1; i < frequency.Length; i++)
+            frequency[i] += frequency[i - 1];
+
+        // constructing the sorted array based on the freqency of each element in ascending order
+        int[] sortedArr = new int[arr.Length];
+        for (int i = 0; i < arr.Length; i++) {
+            sortedArr[frequency[arr[i]] - 1] = arr[i];
+            frequency[arr[i]]--;
+        }
+
```
Comment (aashutoshrathi) [inline counting_sort/CountingSort.cs]: no need of blank line here

File: counting_sort/CountingSort.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return sortedArr;
+    }
+
+    public static void Main()
+    {
+        int[] arr = new int[10000];
+        Random rand = new Random();
+        for (int i = 0; i < arr.Length; i++)
```
Comment (aashutoshrathi) [inline counting_sort/CountingSort.cs]: Check Counting sort in other languages, try not to use Random and hardcode the input values.
https://github.com/iiitv/algos/blob/63ac50bea5371eecf86bcc422ccd3f44eb394ec9/counting_sort/counting_sort.c#L32

**Pull request merged**